ui/src/treeDataProvider/AzureNode.ts

```diff
@@ -88,8 +88,10 @@ export class AzureNode<T extends IAzureTreeItem = IAzureTreeItem> implements IAz
         await this.treeDataProvider.refresh(this);
     }
 
-    public openInPortal(): void {
-        (<(s: string) => void>opn)(`${this.environment.portalUrl}/${this.tenantId}/#resource${this.id}`);
+    public openInPortal(id?: string): void {
+        id ?
```

Comment:
You can simplify by doing one of the following:
`id = id === undefined ? this.id : id`
`id = id || this.id`

That way you don't have duplicate the open and url logic

Reply:
This is why you are "boss".